### PR TITLE
Feature/connection manager logs

### PR DESCRIPTION
**Pull request by Aprogiena**
introduce logs to connection classes

### CODE REVIEW

**Comment by bokobza** [comment]
I'll have a look at this.

File: Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs [hunk 1]
```diff
@@ -54,29 +40,31 @@ protected override void AttachCore()
...
+
+			if ((node.State == NodeState.Failed) || (node.State == NodeState.Offline))
 			{
-			    this.logger.LogInformation("Node " + node.RemoteSocketEndpoint + " offline");
-				if(node.DisconnectReason != null && !string.IsNullOrEmpty(node.DisconnectReason.Reason))
-				    this.logger.LogInformation("Reason: " + node.DisconnectReason.Reason);
-				this.ConnectionManager.RemoveConnectedNode(node);
+			    this.logger.LogInformation("Node {0} offline.", node.RemoteSocketEndpoint);
+
+                if (node.DisconnectReason != null && !string.IsNullOrEmpty(node.DisconnectReason.Reason))
```
Comment (bokobza) [inline Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs]: () around node.DisconnectReason != null.
Although I don't really mind it myself.

File: Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs [hunk 2]
```diff
@@ -54,29 +40,31 @@ protected override void AttachCore()
...
+				    this.logger.LogInformation("Reason: {0}", node.DisconnectReason.Reason);
+
+                this.ConnectionManager.RemoveConnectedNode(node);
```
Comment (bokobza) [inline Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs]: tabs vs spaces formatting

File: Stratis.Bitcoin/Connection/DropNodesBehaviour.cs [hunk 3]
```diff
@@ -23,18 +23,18 @@ public DropNodesBehaviour(ConcurrentChain chain, IConnectionManager connectionMa
             this.chain = chain;
             this.connection = connectionManager;
 
-            // 80% of current max connections, the last 20% will only 
-            // connect to nodes ahead of the current best chain
+            // 80 % of current max connections, the last 20 % will only
```
Comment (bokobza) [inline Stratis.Bitcoin/Connection/DropNodesBehaviour.cs]: I think % attached to the number is fine.

File: Stratis.Bitcoin/Connection/ConnectionManager.cs [hunk 4]
```diff
@@ -42,24 +42,30 @@ public interface IConnectionManager : IDisposable
...
-        // The maximum number of entries in an 'inv' protocol message
-        public const int MAX_INV_SZ = 50000;
+        // The maximum number of entries in an 'inv' protocol message.
+        public const int MaxInventorySize = 50000;
+
+        private readonly ILogger logger;
+        private readonly INodeLifetime nodeLifetime;
 
         private readonly NodesCollection connectedNodes = new NodesCollection();
+        public IReadOnlyNodesCollection ConnectedNodes { get { return this.connectedNodes; } }
```
Comment (bokobza) [inline Stratis.Bitcoin/Connection/ConnectionManager.cs]: please use` =>` for these.

File: Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs [hunk 5]
```diff
@@ -25,6 +31,7 @@ public class ConnectionManagerBehavior : NodeBehavior
         public ConnectionManagerBehavior(bool inbound, IConnectionManager connectionManager, ILoggerFactory loggerFactory)
 		{
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName, $"[{this.GetHashCode():x}] ");
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
```
Comment (bokobza) [inline Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs]: Did you mean this to be infoLogger?

File: Stratis.Bitcoin/Connection/DropNodesBehaviour.cs (hunk 3 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Connection/DropNodesBehaviour.cs]: hmm, you are right https://english.stackexchange.com/a/3284 - english does not prescribe this
in my language there is a notation that "80%" means something else than "80 %"
i.e. it is correct to write "80% solution of water with whatever" but not "80 % solution of water with whatever" and vice versa, it is correct to write "80 % of all people do something" but not "80% of all people do something"

but english seems to be weird in that, so :/

Comment (Aprogiena) [reply Stratis.Bitcoin/Connection/DropNodesBehaviour.cs]: do you want me to revert that ?

Comment (bokobza) [reply Stratis.Bitcoin/Connection/DropNodesBehaviour.cs]: thanks for the explanation :-D
yes please :-)

File: Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs (hunk 5 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs]: fixed tx

File: Stratis.Bitcoin/Connection/DropNodesBehaviour.cs (hunk 3 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Connection/DropNodesBehaviour.cs]: fixed back

File: Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs (hunk 1 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs]: improved

File: Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs (hunk 2 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs]: i will soon probably create PR to untabify whole solution

Comment (Aprogiena) [reply Stratis.Bitcoin/Connection/ConnectionManagerBehavior.cs]: but fixed here

File: Stratis.Bitcoin/Connection/ConnectionManager.cs (hunk 4 above)
Comment (Aprogiena) [reply Stratis.Bitcoin/Connection/ConnectionManager.cs]: let's leave this as is now, but discuss this in #688

Comment (bokobza) [reply Stratis.Bitcoin/Connection/ConnectionManager.cs]: ok

**Pull request merged**